mtags/src/main/scala-3/scala/meta/internal/pc/completions/MatchCaseCompletions.scala

```diff
@@ -145,17 +143,22 @@ object CaseKeywordCompletion:
       )
     else
       // Step 1: walk through scope members.
+      def isValid(sym: Symbol) = !parents.isParent(sym)
+        && (sym.is(Case) || sym.is(Flags.Module) || sym.isClass)
+        && parents.isSubClass(sym, false)
       indexedContext.scopeSymbols
-        .foreach(s => visit(s.info.dealias.typeSymbol, s.decodedName, Nil))
+        .map(_.info.dealias.typeSymbol)
+        .filter(isValid(_))
+        .foreach(s => visit(s, s.decodedName, Nil))
```

Comment:
Personally, I prefer to avoid the usage of transformers, to end with a `foreach`, and you iterate over your collection 3 times.

```suggestion
        .foreach(s => {
          val ts  = s.info.dealias.typeSymbol
          if (isValid(ts))
            visit(ts, ts.decodedName, Nil)
         )}
```

Reply:
Thank, you are right, it's better this way